cmd/lncli/main.go

```diff
@@ -161,6 +142,53 @@ func getClientConn(ctx *cli.Context, skipMacaroons bool) *grpc.ClientConn {
 	return conn
 }
 
+// parseArgs parses the TLS certificate and macaroon paths from the command.
+func parseArgs(ctx *cli.Context) (string, string) {
+	// We'll start off by parsing the active chain and network. These are
+	// needed to determine the correct path to the macaroon when not
+	// specified.
+	chain := strings.ToLower(ctx.GlobalString("chain"))
+	switch chain {
+	case "bitcoin", "litecoin":
+	default:
+		err := fmt.Errorf("unknown chain: %v", chain)
+		fatal(err)
+	}
+
+	network := strings.ToLower(ctx.GlobalString("network"))
+	switch network {
+	case "mainnet", "testnet", "regtest", "simnet":
+	default:
+		err := fmt.Errorf("unknown network: %v", network)
+		fatal(err)
+	}
+
+	var tlsCertPath, macPath string
```

Comment:
The current implementation returns an empty `tlsCertPath` and `macPath` in the case of `defaultLndDir`. We have to assign the defaults here like this:
```
tlsCertPath := filepath.Join(defaultLndDir, defaultTLSCertFilename)
macPath := filepath.Join(
  defaultLndDir, defaultDataDir, defaultChainSubDir,
  chain, network, defaultMacaroonFilename,
)
```
Otherwise this function will return an empty `tlsCertPath` and error `lncli` with the following:
```
➜  lnd git:(macaroons-by-network) ✗ lncli create
[lncli] open : no such file or directory
```

Furthermore, I think the whole conditional `if lndDir != defaultLndDir {` is redundant. If we remove the default value from `tlscertpath` we can do:
```
tlsCertPath := cleanAndExpandPath(ctx.GlobalString("tlscertpath"))
if tlsCertPath == "" {
	tlsCertPath = filepath.Join(lndDir, defaultTLSCertFilename)
}
macPath := cleanAndExpandPath(ctx.GlobalString("macaroonpath"))
if macPath == "" {
	macPath = filepath.Join(
		lndDir, defaultDataDir, defaultChainSubDir,
		chain, network, defaultMacaroonFilename,
	)
}
```

Reply:
I'm also experiencing this
```
# lncli unlock
[lncli] open : no such file or directory
```